_source/_posts/2022-09-26-java-jetty-oauth.md

```diff
@@ -0,0 +1,1017 @@
+import java.util.List;
+
+@Controller
+@RequestMapping(path = "/hikes")
+public class WebController {
+
+   private List<String> hikes = new ArrayList<>(Arrays.asList(
+            "Wonderland Trail", "South Maroon Peak", "Tour du Mont Blanc",
+            "Teton Crest Trail", "Everest Base Camp via Cho La Pass", "Kesugi Ridge"
+    ));
+
+    @GetMapping("")
+    @ResponseBody
+    public String indexGet() {
+        return String.join("\n", this.hikes);
+    }
+
+    @PostMapping("")
+    @ResponseBody
+    public String indexPost(@RequestParam String hike, HttpServletResponse response) {
+        if (hike == null) {
+            response.setStatus(400);
+            return "Param 'hike' cannot be null.";
+        }
+        else if (this.hikes.contains(hike)) {
+            response.setStatus(400);
+            return "The hike '"+hike+"' already exists.";
+        }
+        else {
+            this.hikes.add(hike);
+            return String.join("\n", this.hikes);
+        }
+    }
+
+    @DeleteMapping("")
+    @ResponseBody
+    public String indexDelete(@RequestParam String hike, HttpServletResponse response) {
+        if (hike == null) {
+            response.setStatus(400);
+            return "Param 'hike' cannot be null.";
+        } else {
+            this.hikes.remove(hike);
+            return String.join("\n", this.hikes);
+        }
+    }
+
+}
+```
+
+Some of you might notice that Spring Boot is still using the Java EE Servlet API package and not the newer Jakarta API. It looks like this migration is scheduled for Spring Boot 3.0.0 (see [this issue](https://github.com/spring-projects/spring-boot/issues/31720)).
+
+To run the Spring Boot app, open a Bash shell and navigate to the `spring-boot-jetty-maven` subdirectory.
+
+Run the following.
+
+```bash
+./mvnw spring-boot:run
+```
+
+Wait for it to finish initializing.
+
+```bash
+...
+2022-09-05 16:44:06.968  INFO 1352613 --- [           main] o.s.b.web.embedded.jetty.JettyWebServer  : Jetty started on port(s) 8080 (http/1.1) with context path '/'
+2022-09-05 16:44:06.974  INFO 1352613 --- [           main] com.demo.SpringBootJettyApplication      : Started SpringBootJettyApplication in 0.832 seconds (JVM running for 1.052)
+```
+
+Test it with HTTPie.
+
+```bash
+http :8080/hikes
+```
+
+```
+HTTP/1.1 200 OK
+...
+
+Wonderland Trail
+South Maroon Peak
+Tour du Mont Blanc
+Teton Crest Trail
+Everest Base Camp via Cho La Pass
+Kesugi Ridge
+```
+
+This web service has the same features as the `@WebServlet` version: GET, POST, and DELETE but no PUT.
+
+## Deploy the Spring Boot Project
+
+You now have a Spring Boot application that runs on an embedded Jetty container. To deploy it to a production server, build an executable jar file using `./mvnw package`, copy this jar file (found in the `target` directory) to the server, and run it using java -jar <your jar file name>.jar. There's no need for a separate web server since this jar contains an embedded Jetty web server.
```

Comment:
```suggestion
You now have a Spring Boot application that runs on an embedded Jetty container. To deploy it to a production server, build an executable jar file using `./mvnw package`, copy this jar file (found in the `target` directory) to the server, and run it using `java -jar <your jar file name>.jar`. There's no need for a separate web server since this jar contains an embedded Jetty web server.
```